block_writer/main.go

```diff
@@ -179,38 +188,60 @@ func main() {
 	writers := make([]blockWriter, *concurrency)
 
 	errCh := make(chan error)
+	var wg sync.WaitGroup
 	for i := range writers {
+		wg.Add(1)
 		writers[i] = newBlockWriter(db)
-		go writers[i].run(errCh)
+		go writers[i].run(errCh, &wg)
 	}
 
 	var numErr int
-	for range time.Tick(*outputInterval) {
-		now := time.Now()
-		elapsed := time.Since(lastNow)
-		dumps := atomic.LoadUint64(&numBlocks)
-		fmt.Printf("%6s: %6.1f/sec",
-			time.Duration(time.Since(start).Seconds()+0.5)*time.Second,
-			float64(dumps-lastNumDumps)/elapsed.Seconds())
-		if numErr > 0 {
-			fmt.Printf(" (%d total errors)\n", numErr)
-		}
-		fmt.Printf("\n")
-		for {
-			select {
-			case err := <-errCh:
-				numErr++
-				if !*tolerateErrors {
-					log.Fatal(err)
-				} else {
-					log.Print(err)
-				}
-				continue
-			default:
+	tick := time.Tick(*outputInterval)
```

Comment:
This method never returns as we `os.Exit(0)` below. And you can't actually clean up a `time.Tick` channel.